numba/parfors/parfor.py

```diff
@@ -615,6 +615,27 @@ def dump(self, file=None):
             block.dump(file)
         print(("end parfor {}".format(self.id)).center(20, '-'), file=file)
 
+    def validate_params(self, typemap):
+        """
+        Check that Parfors params are of valid types.
+        """
+        if self.params is None:
+            msg = ("Cannot run parameter validation on a Parfor with params"
+                   "not set")
+            raise ValueError(msg)
+        for p in self.params:
+            ty = typemap.get(p)
+            if ty is None:
+                msg = ("Cannot validate parameter %s, there is no type "
+                       "information available")
+                raise ValueError(msg)
```

Comment:
@DrTodd13 any idea if this can occur in practice and if so how to trigger these conditions? `Parfors.params` is written to just prior to lowering, I don't know how this situation would practically arise but I can just mock it as needed.

Reply:
Should never happen.  Don't know how to force a test.